test/unit/org/apache/cassandra/cql3/CQLTester.java

```diff
@@ -1969,24 +1981,82 @@ private static boolean isEmptyContainerNull(DataType type,
      * @param source the source TableMetadata
      * @param target the target TableMetadata
-     * @param compareParams wether compare table params
+     * @param compareParams wether compare table's params
      * @param compareIndexes wether compare table's indexes
-     * @param compareTrigger wether compare table's triggers
+     * @param compareTriggers wether compare table's triggers
      * */
-    protected boolean equalsWithoutTableNameAndDropCns(TableMetadata source, TableMetadata target, boolean compareParams, boolean compareIndexes, boolean compareTrigger)
+    protected boolean equalsWithoutTableNameAndDropCns(TableMetadata source, TableMetadata target, boolean compareParams, boolean compareIndexes, boolean compareTriggers)
     {
         return source.partitioner.equals(target.partitioner)
                && source.kind == target.kind
                && source.flags.equals(target.flags)
                && (!compareParams || source.params.equals(target.params))
-               && (!compareIndexes || source.indexes.equals(target.indexes))
-               && (!compareTrigger || source.triggers.equals(target.triggers))
+               && compareIndexesOrTriggerWithoutName(compareIndexes,
+                                                     compareTriggers,
+                                                     source.indexes,
+                                                     target.indexes,
+                                                     source.triggers,
+                                                     target.triggers,
+                                                     target.keyspace,
+                                                     target.name)
                && columnsEqualWitoutKsTb(source, target);
     }
 
+    /**
+     * Compare the source table and target table's indexes and triggers without compare their name,
+     * because the names of the target indexes and triggers are randomly genrated base on the source
+     * table name and source keyspace name with a rule thant can see
+     * {@link KeyspaceMetadata#findAvailableTriggerName(String)} and {@link KeyspaceMetadata#findAvailableIndexName(String)}
+     * for more detail.
+     * @param compareIndexes wether compare indexes or not
+     * @param compareTriggers wether compare triggers or not
+     * @param sourceIdxes source indexes
+     * @param targetIdxes target indexes
+     * @param sourceTriggers source triggers
+     * @param targetTriggers target triggers
+     * @return true if source table's indexes or triggers are same with target table's indexes or triggers without compare keyspace and table name
+     * */
+    private boolean compareIndexesOrTriggerWithoutName(boolean compareIndexes, boolean compareTriggers, Indexes sourceIdxes, Indexes targetIdxes, Triggers sourceTriggers, Triggers targetTriggers, String targetKs, String targetTb)
```

Comment:
Similar nit. I think splitting this method into indexes and triggers may help with the method signature.